code/modules/projectiles/guns/syringe_gun.dm

```diff
@@ -95,3 +94,285 @@
+
+	else if(istype(A, /obj/item/reagent_containers/syringe))
+		insert_single_syringe(A, user)
+
+	else if(istype(A, /obj/item/reagent_containers))
+		// Loading with chemicals (but not from syringes)
+		var/obj/item/reagent_containers/incoming = A
+		if(!incoming.is_drainable())
+			return
+
+		if(!incoming.reagents.total_volume)
+			to_chat(user, "<span class='warning'>[incoming] is empty!</span>")
+			return
+
+		if(reagents.holder_full())
+			to_chat(user, "<span class='warning'>[src]'s internal reservoir is full!</span>")
+			return
+
+		var/trans = incoming.reagents.trans_to(src, incoming.amount_per_transfer_from_this)
+		to_chat(user, "<span class='notice'>You transfer [trans] unit\s of the solution to [src]'s internal reservoir.</span>")
+		update_loaded_syringe()
+
+		// Reset the reservoir alarm
+		alarmed = FALSE
+
+	else
+		return ..()
+
+// Allow for emptying your deathmix, or for sec to find out what you were dumping into people
+/obj/item/gun/syringe/rapidsyringe/afterattack(atom/target, mob/living/user, flag, params)
+	if(istype(target, /obj/item/reagent_containers) && !istype(target, /obj/item/reagent_containers/syringe))
+		var/obj/item/reagent_containers/destination = target
+
+		if(!destination.is_refillable())
+			return
+
+		if(!reagents.total_volume)
+			to_chat(user, "<span class='warning'>[src]'s internal reservoir is empty!</span>")
+			return
+
+		if(destination.reagents.holder_full())
+			to_chat(user, "<span class='warning'>[destination] is full.</span>")
+			return
+
+		var/transfer_source = "[src]'s internal reservoir"
+		var/transfer_amount
+
+		// Empty directly from the syringe, if it's loaded.
+		// Otherwise, we wouldn't be able to fully empty the gun's chems without firing it.
+		if(chambered?.BB && chambered.BB.reagents.total_volume)
+			transfer_amount = chambered.BB.reagents.trans_to(destination, get_units_per_shot())
+			transfer_source = "the syringe chambered in [src]"
+		else
+			transfer_amount = reagents.trans_to(destination, reservoir_transfer_amount)
+
+		to_chat(user, "<span class='notice'>You drain [transfer_amount] unit\s from [transfer_source] into [destination].</span>")
+		// Refill the syringe
+		update_loaded_syringe()
+		return TRUE
+	else
+		return ..()
+
+// Switch the amount of reagents used per shot.
+/obj/item/gun/syringe/rapidsyringe/AltClick(mob/living/user)
+
+	transfer_amount_selection++
+	if(transfer_amount_selection > length(possible_transfer_amounts))
+		transfer_amount_selection = 1
+
+	playsound(src, 'sound/weapons/gun_interactions/selector.ogg', 25, 1)
+	to_chat(user, "<span class='notice'>[src] will now fill each syringe with up to [get_units_per_shot()] units.</span>")
+	update_loaded_syringe()
+
+/// Play an alert when the reservoir has run out of reagents so people don't unknowingly dump empty syringes into their foes
+/// Running out of syringes is just handled by *click*
+/obj/item/gun/syringe/rapidsyringe/afterattack(atom/target, mob/living/user, flag, params)
```

Comment:
You have two `afterattack` procs for this syringe gun.

Reply:
i am a competent coder i promise